### PR TITLE
Replacing EasyMDE with regular CodeMirror

### PR DESCRIPTION
**Pull request by jcontonio**
* Adds latest CodeMirror
* Should behave exactly as it did before, minus background colors on code
* Adds syntax highlighting for html, javascript, css, etc

We can add background colors later with our own custom AddOn

### CODE REVIEW

**Comment by jcontonio** [comment]
PS. Make sure you test the Format menu

File: web/src/client/js/components/FieldText/FieldTextComponent.js [hunk 1]
```diff
@@ -1,58 +1,70 @@
-import React, { useEffect, useRef, useState } from 'react'
+import React, { useEffect, useRef } from 'react'
 import PropTypes from 'prop-types'
-import EasyMDE from 'easymde'
+import CodeMirror from 'codemirror/lib/codemirror'
 
-import './EasyMDE.scss'
+// import 'codemirror/addon/mode/overlay'
+// import 'codemirror/mode/xml/xml'
+// import 'codemirror/mode/markdown/markdown'
```
Comment (jjidt) [inline web/src/client/js/components/FieldText/FieldTextComponent.js]: wanna leave these commented out lines?

File: web/src/client/js/components/FieldText/FieldTextComponent.js [hunk 2]
```diff
@@ -1,58 +1,70 @@
...
+
+import './codemirror.css'
 import './FieldText.scss'
 
+window.CodeMirror = CodeMirror
+
 const FieldTextComponent = ({ autoFocusElement, field, onBlur, onChange, onFocus, readOnly }) => {
   const textRef = useRef()
-  const [editor, setEditor] = useState(null)
+  const editorRef = useRef()
```
Comment (jjidt) [inline web/src/client/js/components/FieldText/FieldTextComponent.js:19]: nice

**Review by jjidt — CHANGES_REQUESTED** [review]
A few things:

Screen Shot 2020-01-23 at 1 18 24 PM

The cursor is expanding to all lines on line wrap

Format menu issues:

- Adding Bold element initially adds a horizontal rule
- Link and horizontal rule items don't add element or close menu
- Ordered list and Unordered list are reversed

**Review by jjidt — APPROVED** [review]
👍 nice

**Pull request merged**